checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -2121,10 +2122,13 @@ private void propagateObligationsToSuccessorBlock(
         CFStore mcStore;
         AccumulationStore cmStore;
         if (currentBlockNodes.size() == 0 /* currentBlock is special or conditional */) {
-          cmStore =
-              obligationGoesOutOfScopeBeforeSuccessor
-                  ? analysis.getInput(currentBlock).getRegularStore() // 1a. (CM)
-                  : regularStoreOfSuccessor; // 1b. (CM)
+          cmStore = getStoreForEdgeFromEmptyBlock(currentBlock, successor); // 1b. (CM)
+          // TODO we would like to get the store for the specific control-flow edge that the
+          // Must Call Checker computed as well.  However, this does not seem to be possible with
+          // current APIs.  The MustCallAnnotatedTypeFactory keeps a flowResult field that
+          // internally holds the relevant TransferInput objects in its stores map, but it does not
+          // expose these objects via its API.  TransferInput objects keep the "then store" and
+          // "else store" information that we need
```

Comment:
Specifically, the APIs we need in `AnalysisResult` would be something like `getThenStoreBefore(Block)` and `getElseStoreBefore(Block)`, so we could write code similar to what is written starting at line 2184.  @smillst @mernst do you think these would be reasonable APIs to add?

Reply:
I decided to just punt on this for now, as I cannot think of a realistic test case that expose any imprecision.